src/preload.py

```diff
@@ -0,0 +1,342 @@
+    env = {v["name"]: v["value"] for v in data.get("environment_variable", [])}
+    return {
+        "appId": data["id"],
+        "name": data["app_name"],
+        "commit": data["commit"],
+        "imageRepo": image_repo,
+        "imageId": image_id,
+        "env": {k: v for k, v in env.items() if not k.startswith("RESIN_")},
+        "config": {k: v for k, v in env.items() if k.startswith("RESIN_")},
+    }
+
+
+def get_container_size(image_repo):
+    log.info("Fetching container size for {}".format(image_repo))
+    token = get_registry_token(image_repo)
+    data = registry("v2/{}/manifests/latest".format(image_repo), token)
+    headers = {"Range": "bytes=-4"}
+    size = 0
+    # the last 4 bytes of each gzipped layer are the layer size % 32
+    for layer in data["fsLayers"]:
+        endpoint = "v2/{}/blobs/{}".format(image_repo, layer["blobSum"])
+        layer_size_bytes = registry(
+            endpoint,
+            token,
+            headers,
+            decode_json=False
+        )
+        layer_size = int.from_bytes(layer_size_bytes, byteorder="little")
+        size += layer_size
+        log.info("{} {}".format(endpoint, size))
+    log.info("Container size: {}".format(human_size(size)))
+    return size
+
+
+def get_resin_os_version(image):
+    with mount_context_manager(image, 2) as mountpoint:
+        with open(os.path.join(mountpoint, "etc/os-release")) as f:
+            for line in f:
+                key, value = line.split("=", 1)
+                if key == "VERSION":
+                    return value.strip('"\n"')
+
+
+def expand_image(image, additional_space):
+    log.info("Expanding image size by {}".format(human_size(additional_space)))
+    # Add zero bytes to image to be able to resize partitions
+    with open(image, "a") as f:
+        size = f.tell()
+        f.truncate(size + additional_space)
+
+
+def expand_partitions(image):
+    """Resizes partitions 4 & 6 to the end of the image"""
+    log.info(
+        "Expanding extended partition 4 and logical partition 6 to the end of "
+        "the disk image."
+    )
+    parted("-s", image, "resizepart", 4, "100%", "resizepart", 6, "100%")
+
+
+def expand_ext4(image):
+    """Resizes the ext4 filesystem in the 6th partition of the image"""
+    with losetup_context_manager(image, 6) as loop_device:
+        log.info("Using {}".format(loop_device))
+        log.info("Resizing filesystem")
+        try:
+            status = e2fsck("-p", "-f", loop_device, _ok_code=[0, 1, 2])
+            status = status.exit_code
+            if status == 0:
+                log.info("e2fsck: File system OK")
+            else:
+                log.warning("e2fsck: File system errors corrected")
+        except ErrorReturnCode:
+            log.error("e2fsck: File system errors could not be corrected")
+            exit(1)
+        resize2fs("-f", loop_device)
+
+
+def expand_btrfs(mountpoint):
+    # For btrfs we'll need to mount the fs.
```

Comment:
This comment seems to be in the wrong place, as the line below it isn't doing what the comment says ;)